### ISSUE TITLE
Cannot import AutoIt Library into Ride

### ISSUE THREAD

**Issue by Acetylen3** [nokia/robotframework-autoitlibrary#15]
I'm using Windows 10, Python 2.7.
When I try to deploy Ride to run robot tests on Windows 10, I found Ride cannot import keywords from AutoItLibrary, but Ride is able to run test with AutoItLibrary Keywords.
I used pip in Python 2.7 to install everything.

**Comment by lucyking** [comment]
Could you run ```ride.py``` in cmd console and reproduce this issue, then paste the output log here?

**Comment by Acetylen3** [comment]
> Could you run `ride.py` in cmd console and reproduce this issue, then paste the output log here?

20190325 10:13:11.110 [INFO]: Found Robot Framework version 3.1.1 from C:\Python27\lib\site-packages\robot.

20190325 10:13:11.110 [INFO]: Started RIDE 1.7.3.1 using python version 2.7.15 with wx version 4.0.4 in win32.

20190325 10:39:21.889 [WARN]: Importing test library "AutoItLibrary" failed

Traceback (most recent call last):
Initializing test library 'AutoItLibrary' with no arguments failed: RobotNotRunningError: Cannot access execution context
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\robotide\lib\robot\running\testlibraries.py", line 167, in _get_instance
    return libcode(*self.positional_args, **dict(self.named_args))
  File "C:\Python27\lib\site-packages\AutoItLibrary\__init__.py", line 88, in __init__
    self._OutputDir  = self._get_log_dir()
  File "C:\Python27\lib\site-packages\AutoItLibrary\__init__.py", line 182, in _get_log_dir
    variables = BuiltIn().get_variables()
  File "C:\Python27\lib\site-packages\robot\libraries\BuiltIn.py", line 1293, in get_variables
    return self._variables.as_dict(decoration=is_falsy(no_decoration))
  File "C:\Python27\lib\site-packages\robot\libraries\BuiltIn.py", line 75, in _variables
    return self._namespace.variables
  File "C:\Python27\lib\site-packages\robot\libraries\BuiltIn.py", line 71, in _namespace
    return self._get_context().namespace
  File "C:\Python27\lib\site-packages\robot\libraries\BuiltIn.py", line 66, in _get_context
    raise RobotNotRunningError('Cannot access execution context')
  File "C:\Python27\lib\site-packages\robotide\spec\librarymanager.py", line 87, in _fetch_keywords
    return get_import_result(path, library_args)
  File "C:\Python27\lib\site-packages\robotide\spec\libraryfetcher.py", line 24, in get_import_result
    lib = robotapi.TestLibrary(path, args)
  File "C:\Python27\lib\site-packages\robotide\lib\robot\running\testlibraries.py", line 54, in TestLibrary
    lib.create_handlers()
  File "C:\Python27\lib\site-packages\robotide\lib\robot\running\testlibraries.py", line 100, in create_handlers
    self._create_handlers(self.get_instance())
  File "C:\Python27\lib\site-packages\robotide\lib\robot\running\testlibraries.py", line 159, in get_instance
    self._libinst = self._get_instance(self._libcode)
  File "C:\Python27\lib\site-packages\robotide\lib\robot\running\testlibraries.py", line 169, in _get_instance
    self._raise_creating_instance_failed()
  File "C:\Python27\lib\site-packages\robotide\lib\robot\running\testlibraries.py", line 314, in _raise_creating_instance_failed
    % (self.name, args_text, msg, details))

**Comment by lucyking** [comment]
1. Could you using python 3.7.x if possible?
2. Try manually include  BuiltIn lib first: 
```
*** Settings ***
Library           BuiltIn
Library           AutoItLibrary
```

**Comment by Acetylen3** [comment]
> 1. Could you using python 3.7.x if possible?
> 2. Try manually include  BuiltIn lib first:
> 
> ```
> *** Settings ***
> Library           BuiltIn
> Library           AutoItLibrary
> ```
1. I will try to migrate to Python 3.7, that means AutoItLibrary is dropping support to Python 2.7?
2. I have tried but it give the same error as above in the log.

**Comment by lucyking** [comment]
@Acetylen3 It support Python 2.7 in fact. Just try to confirm that there's no relation between python version and this issue.

**Comment by Acetylen3** [comment]
@lucyking I keep using Python 2.7 is just because of dependence issue with other Robot Library.

**Comment by lucyking** [comment]
Migration always takes a lot of time. keeping the version seems better.
then, could you provide a demo.robot which can reproduce the issue?

**Comment by Acetylen3** [comment]
> Migration always takes a lot of time. keeping the version seems better.
> then, could you provide a demo.robot which can reproduce the issue?

I can reproduce the issue even with a empty robot test script:
*** Settings ***
Library           BuiltIn
Library           AutoItLibrary  <---- Highlighted in red

*** Test Cases ***
case 1
    BuiltIn.Sleep    1s

Did the Traceback provide any useful information?

**Comment by lucyking** [comment]
Of course, your simple demo make sense to solving the issue.  
I'll try to set up a py2.x env and reproduce the issue next.

**Comment by Acetylen3** [comment]
> Of course, your simple demo make sense to solving the issue.
> I'll try to set up a py2.x env and reproduce the issue next.

I can get around this issue by downgrading robotframework-autoitlibrary from v1.2.3 to v1.2.2
From the stack-trace, the problem seems to be caused by calling robot builtin library without under robot execution context. 
I hope it may help.

**Comment by lucyking** [comment]
I've fixed this issue, which instroduced by v1.2.3 indeed. Pls update to v1.2.4, everything goes well.

**Comment by remicharleslageixarena** [comment]
Hi everybody,

I have the same issue.
In robotframework , the AutoitLibrary  (is Highlighted in red)

I use python 3.7.
The installation of library AutoItLibrary  with pip command was succesfully without any error.
(robotframework-autoitlibrary   1.2.7)
But robotframework can not see it.

in tools->ride.log 
"20201117 13:25:52.641 [INFO]: Found Robot Framework version 3.2.1 from C:\Python37\lib\site-packages\robot.
20201117 13:25:52.641 [INFO]: Started RIDE 1.7.4.2 using python version 3.7.7 with wx version 4.0.7.post2 in win32.
20201117 13:26:06.969 [WARN]: Importing test library "AutoItLibrary" failed"

Can you help me?
Thanks in advance


I resolve my probleme.

becareful  : open  'cmd' with administrator rights is not the same to open a session with administrator account on windows 10.
I connected with admnistrator account and I re-installed   the library   ...."pip   install robotframework-autoitlibrary==1.2.5"
and now it works fine.
